### ISSUE TITLE
Notebook.ai Community Drive

### ISSUE THREAD

**Issue by drusepth** [indentlabs/notebook#198]
TL;DR: A community drive and referral program to shoot Notebook.ai up to 350 subscribers. For more info, check out [the unpublished page about it](https://www.notebook.ai/for/friends) (so please don't share yet). 

What's left to do:
- [x] Review /for/friends landing page with @amymartika
- [x] Generate referral links for all users
- [x] Expose referral link on page CTA modal
- [x] Integrate referral link in sign-up process
- [x] Store referrals under UserReferrals to credit both users on premium subs
- [x] Finish sharing modal on `/for/friends`
- [x] Code voting system for community features
  - Community challenges, #195 
  - Universe collaboration, #25 
  - Forums & chat rooms, #196 
  - Expert Q&A sessions, #197 
- [x] Write FAQ answers

Final steps:

- [x] OK referral page for logged-out users
- [x] Add in click-to-tweet links to sharing modal
- [x] Award votes and storage space on subscription change
- [x] Make sure referrals landing page looks good when shared to social networks
- [x] Write blog post
- [x] Write email
- [x] Add banner to site
- [x] Update drive end-date to ~1.5 months from start
- [x] Go live

Post-release:

- [x] Assign all users a referral code
  - `User.all.map(&:initialize_referral_code)`
- [x] Assign all users a secure code
  - `User.all.map(&:initialize_secure_code)`
- [x] Give all existing Premium users a feature vote
  - `User.where(selected_billing_plan_id: [3, 4, 5, 6]).map { |user| user.votes.create }`
- [x] Give all existing Premium users a raffle entry
  - `User.where(selected_billing_plan_id: [3, 4, 5, 6]).map { |user| user.raffle_entries.create }`
- [x] Initialize votables
  - `heroku rake voting:create_votables`

**Comment by drusepth** [comment]
Everything's soft launched and live now. Still need to send out emails and post the blog post.

**Comment by drusepth** [comment]
Community drive has concluded.
